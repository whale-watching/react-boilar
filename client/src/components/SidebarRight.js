import React from 'react';
import Close from 'react-icons/lib/md/close';
import { IncomingMessage } from 'http';

const colors = {
  violet: '#746db4',
  light: '#dedef7',
  dark: '#263238',
  pink: '#ff95c7',
  green: '#92db25'
};

class SidebarRight extends React.Component {

  closeSidebar = () => {
    const sidebar = document.getElementsByClassName('sidebar-right')[0];

    if (sidebar.className.match(/(?:^|\s)sidebar-right-open(?!\S)/)) {
      sidebar.classList.remove('sidebar-right-open');
      sidebar.classList.add('sidebar-right-closed');
    }
  }

  changeTheme = (e) => {
    const theme = e.target.className;

    document.getElementById('theme').href = `/dist/${theme}.css`;
  }

  changeBackground = (e) => {
    const background = e.target.className;
    const chatContent = document.getElementsByClassName('chat-content')[0];
    
    chatContent.style.backgroundImage = `url('/img/${background}.png')`;
  }

  render() {
    return (
      <div className="sidebar-right sidebar-right-closed">
        <div className="close">
          <button onClick={this.closeSidebar}>
            <Close className="icon" size="24px" />
          </button>
        </div>
        <div className="themes">
          <div className="panel">
            <p>Choose Theme</p>
          </div>
          <div className="colors">
            {Object.keys(colors).map((color) => {
              return <div 
                className={color} 
                style={{'background': colors[color]}} 
                onClick={this.changeTheme}>
              </div>
            })}
          </div>
        </div>
        <div className="backgrounds">
          <div className="panel">
            <p>Choose Background</p>
          </div>
          <div className="images">
            <img className="sun" src={'/img/sun.png'} onClick={this.changeBackground} />
            <img className="triangular_white" src={'/img/triangular_white.png'} onClick={this.changeBackground} />
            <img className="triangular_grey" src={'/img/triangular_grey.png'} onClick={this.changeBackground} />
            <img className="flowers" src={'/img/flowers.png'} onClick={this.changeBackground} />
            <img className="triangular_black" src={'/img/triangular_black.png'} onClick={this.changeBackground} />
          </div>
        </div>
      </div>
    )
  }
};

export default SidebarRight;