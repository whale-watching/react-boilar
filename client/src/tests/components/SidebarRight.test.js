import React from 'react';
import { shallow } from 'enzyme';
import SidebarRight from '../../components/SidebarRight';

describe('<SidebarRight />', () => {
  it('renders SidebarRight correctly', () => {
    const wrapper = shallow(<SidebarRight />);

    expect(wrapper).toMatchSnapshot();
  });
});