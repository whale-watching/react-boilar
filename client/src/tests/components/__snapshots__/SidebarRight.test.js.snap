// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarRight /> renders SidebarRight correctly 1`] = `
<div
  className="sidebar-right sidebar-right-closed"
>
  <div
    className="close"
  >
    <button
      onClick={[Function]}
    >
      <MdClose
        className="icon"
        size="24px"
      />
    </button>
  </div>
  <div
    className="themes"
  >
    <div
      className="panel"
    >
      <p>
        Choose Theme
      </p>
    </div>
    <div
      className="colors"
    >
      <div
        className="violet"
        onClick={[Function]}
        style={
          Object {
            "background": "#746db4",
          }
        }
      />
      <div
        className="light"
        onClick={[Function]}
        style={
          Object {
            "background": "#dedef7",
          }
        }
      />
      <div
        className="dark"
        onClick={[Function]}
        style={
          Object {
            "background": "#263238",
          }
        }
      />
      <div
        className="pink"
        onClick={[Function]}
        style={
          Object {
            "background": "#ff95c7",
          }
        }
      />
      <div
        className="green"
        onClick={[Function]}
        style={
          Object {
            "background": "#92db25",
          }
        }
      />
    </div>
  </div>
  <div
    className="backgrounds"
  >
    <div
      className="panel"
    >
      <p>
        Choose Background
      </p>
    </div>
    <div
      className="images"
    >
      <img
        alt="sun"
        className="sun"
        onClick={[Function]}
        src="/img/sun.png"
      />
      <img
        alt="triangular-white"
        className="triangular_white"
        onClick={[Function]}
        src="/img/triangular_white.png"
      />
      <img
        alt="triangular-grey"
        className="triangular_grey"
        onClick={[Function]}
        src="/img/triangular_grey.png"
      />
      <img
        alt="flowers"
        className="flowers"
        onClick={[Function]}
        src="/img/flowers.png"
      />
      <img
        alt="triangular-black"
        className="triangular_black"
        onClick={[Function]}
        src="/img/triangular_black.png"
      />
    </div>
  </div>
</div>
`;
